src/EVM.hs

```diff
@@ -2267,10 +2381,10 @@ finishFrame how = do
               push 0
 
             -- Case 3: Error during a call?
-            FrameErrored _ -> do
+            FrameErrored e -> do
               revertContracts
               revertSubstate
-              assign (#state % #returndata) mempty
+              assign (#state % #returndata) (ConcreteBuf (BS8.pack $ show e))
```

Comment:
why do we need to do this for non reverting errors?

Reply:
This has  been removed, thanks!!